doltpy/core/dolt.py

```diff
@@ -401,107 +311,570 @@ def get_commits(self) -> Mapping[str, DoltCommitSummary]:
                 date = datetime.strptime(line.split(':', maxsplit=1)[1].lstrip(), '%a %b %d %H:%M:%S %z %Y')
             elif current_commit is not None:
                 assert current_commit is not None and date is not None and author is not None
-                result[current_commit] = DoltCommitSummary(current_commit, date, author)
+                result[current_commit] = DoltCommit(current_commit, date, author)
                 current_commit = None
             else:
                 pass
 
         return result
 
-    def get_dirty_tables(self) -> Tuple[Mapping[str, bool], Mapping[str, bool]]:
-        """
-        Returns a tuple of maps, the first element is a map keyed on the names of newly created tables, and a second
-        is keyed on modified tables, with the values being boolean flags to indicate whether changes have been stage for
-        commit.
-        :return:
-        """
-        new_tables, changes = {}, {}
+    # returns a table-like structure
+    def diff(self,
+             commit: str = None,
+             other_commit: str = None,
+             table_or_tables: Union[str, List[str]] = None,
+             data: bool = False,
+             schema: bool = False, # can we even support this?
+             summary: bool = False,
+             sql: bool = False,
+             where: str = None,
+             limit: int = None):
+        switch_count = [el for el in [data, schema, summary] if el]
+        assert len(switch_count) <= 1, 'At most one of delete, copy, move can be set to True'
+
+        if type(table_or_tables) == str:
+            tables = [table_or_tables]
+        else:
+            tables = table_or_tables
+
+        args = ['diff']
+
+        if data:
+            if where:
+                args.extend(['--where', where])
+            if limit:
+                args.extend(['--limit', limit])
+
+        if summary:
+            args.append('--summary')
+
+        if schema:
+            args.extend('--schema')
+
+        if sql:
+            args.append('--sql')
+
+        if commit:
+            args.append(commit)
+        if other_commit:
+            args.append(other_commit)
+
+        if tables:
+            args.append(' '.join(tables))
+
+        self.execute(args)
+
+    def blame(self, table_name: str, rev: str = None):
+        args = ['blame']
+
+        if rev:
+            args.append(rev)
+
+        args.append(table_name)
+        self.execute(args)
+
+    def branch(self,
```

Comment:
I think this method would do better as a set of functions `create_branch`, `copy_branch`,`move_branch`

Reply:
I sorta of agree aesthetically, but I think if we want to match the `CLI`, then we do `command_subcommand`, otherwise we break the principle of least surprise.

I agree that heavy conditional logic that does totally different stuff based on arguments is not ideal...